Created browsing page

miyabitanimchi opened the pull request:
I created the browsing page. 
Now there is a warning like below, I already know why, it's because I put 2 components in the map method on `Browsing.tsx`. 

<img width="408" alt="スクリーンショット 2021-12-15 午前2 04 37" src="https://user-images.githubusercontent.com/69213541/146167601-56554af4-3966-4eac-b04f-6d999ae3fe41.png">

Also I implemented a toggle function for the heart but it's not working yet because each card itself it's a link. (you can see the function works if you comment out the `Link` on the card)

And, for the layout of the cards, because I used `justify-center` and `flex-wrap` the card(s) on a last row goes in the center. Probably it's better it(they) are placed in the left side right?

Let me figure out how we can improve them later 🙏

shiv-chan commented:
Thank you @miyabitanimchi for your hard work!
I'm gonna review the code and test it on my browser.

**Just leave a quick comment.**
- I guess the error message shows up because there is no `key` for maps not because of having two maps according to the message.
- what if place the heart toggle with `absolute` or set `z-index` to make it placed above the card itself? Probably, "event propagation" will happen there.
- why don't you use `grid` for this layout instead?

miyabitanimchi commented:
What I changed and fixed are:

- For the layout, I changed Flexbox to Grid.
- On `Browsing.tsx` I added a function to create a new array to extract data for senpai and kohai, and map the array to arrange the user cards (no warning anymore)
- And now you can toggle the heart! 
- Added "... and more" if there are more than 5 skill tags

Thank you for your advice @shiv-chan 💡
Please review again when you can, thank you!

shiv-chan commented on client/src/common/components/browsingPage/UserCard.tsx:
```diff
@@ -27,32 +27,39 @@ const UserCard = ({
+  const handleLike = (e: React.MouseEvent<SVGAElement>) => {
+    e.preventDefault();
+    setIsLiked((prevState) => !prevState);
+    e.stopPropagation();
```
yay 🎉 🎉  now it's working! Thanks!

shiv-chan commented on client/src/common/components/browsingPage/UserCard.tsx:
```diff
@@ -27,32 +27,39 @@ const UserCard = ({
     >
-      <div className="mobile_l_max:h-48 mobile_l_max:w-72 mobile_l_max:p-2.5 h-56 w-80 mobile_l:h-64 mobile_l:w-96 flex-initial bg-white hover:bg-gray-50 rounded-lg p-5 z-0">
-        <div className="flex justify-between items-start mb-7 mb-l-max:mb-2">
+      <div className="w-full h-60 mobile_l_max:h-48 flex-initial bg-white hover:bg-gray-50 rounded-lg mobile_l_max:p-2.5 p-5 z-0">
```
@miyabitanimchi I liked the colour slightly changes when hovering 💖

shiv-chan reviewed (approved):
Thanks a million, @miyabitanimchi 🙏 
The layout and the code are much nicer and cleaner!

Let me make a comment about the spacing on mobile devices.
As you see in the capture below, each card takes up almost all width of the screen, so looks tight. 
![スクリーンショット 2021-12-15 22 12 14](https://user-images.githubusercontent.com/51708229/146333377-774889ab-2c18-4863-914e-c862d1e44942.png)

I updated this in `feature-browsing-page-kaho` branch and it looks like the below.
![スクリーンショット 2021-12-16 0 10 25](https://user-images.githubusercontent.com/51708229/146333821-a17ecd4c-90f4-4b60-ae40-ca89578e7117.png)

I'm going to make a PR to merge this branch, so please take a look.

Thanks again!
